invenio_rdm_records/contrib/journal/custom_fields.py

```diff
@@ -12,14 +12,181 @@
+        """Hook called after the marshmallow serialization of the record.
+
+        :param data: The dumped record data.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The serialized record data.
+        """
+        return data
+
+    def pre_dump(self, data, original={}, **kwargs):
+        """
+        Hook called before the marshmallow serialization of the record.
+
+        :param data: The record data to dump.
+        :param original: The original record data.
+        :param kwargs: Additional keyword arguments.
+        :returns: The data to dump.
+        """
+        return data
+
+
+class FieldLoader(ABC):
+    def post_load(data, many=False, **kwargs):
+        raise NotImplementedError()
+
+    def pre_load(data, many=False, **kwargs):
+        raise NotImplementedError()
+
+
+class JournalDataciteDumper(FieldDumper):
+    """Dump processor for datacite serialization of 'Journal' custom field."""
+
+    def post_dump(self, data, original={}, **kwargs):
+        """Adds the journal information as a related identifier in the DataCite metadata."""
+        custom_fields = original.get("custom_fields", {})
+        journal_data = custom_fields.get("journal:journal", {})
+        issn = journal_data.get("issn")
+
+        # Only dumps if 'issn' is set.
+        if not issn:
+            return data
+
+        # Serialize journal
+        relationType = "IsPartOf"
+        resourceTypeGeneral = "Collection"
+        relatedIdentifierType = "ISSN"
+
+        serialized_journal = {
+            "relatedIdentifier": issn,
+            "relationType": relationType,
+            "relatedIdentifierType": relatedIdentifierType,
+            "resourceTypeGeneral": resourceTypeGeneral,
+        }
+
+        # Update input data
+        related_identifiers = data.get("relatedIdentifiers", [])
+        related_identifiers.append(serialized_journal)
+        data["relatedIdentifiers"] = related_identifiers
+
+        return data
+
+
+class JournalDublinCoreDumper(FieldDumper):
+    """Dump processor for dublin core serialization of 'Journal' custom field."""
+
+    def post_dump(self, data, original={}, **kwargs):
+        """Adds serialized journal data to the input data under the 'sources' key."""
+        custom_fields = original.get("custom_fields", {})
+        journal_data = custom_fields.get("journal:journal", {})
+        title = journal_data.get("title")
+        volume = journal_data.get("volume")
+        issue = journal_data.get("issue")
+        pages = journal_data.get("pages")
+        year = original.get("metadata", {}).get("publication_date")
+
+        if volume and issue:
+            volume = f"{volume}({issue})"
+
+        if not volume:
+            volume = issue
+
+        parts = [
+            title,
+            volume,
+            pages,
+            "({0})".format(year) if year else None,
```

Comment:
```suggestion
            f"({year})" if year else None,
```
Absolutely not necessary, just a personal preference because I think it's a bit easier on the eye